IPython/utils/text.py

```diff
@@ -561,7 +561,13 @@ def _vformat(self, format_string, args, kwargs, used_args, recursion_depth):
 
                 # eval the contents of the field for the object
                 # to be formatted
-                obj = eval(field_name, kwargs)
+                try:
+                    obj = eval(field_name, kwargs)
+                except NameError:
+                    if supressExceptions:
+                       obj = field_name
```

Comment:
Well that where the "then a miracle occurs" in reply to "Easy enough" came from. It is a rather nasty thing we have here.